app.rb

```diff
@@ -0,0 +1,66 @@
+require './person'
+require './student'
+require './rentals'
+require './teacher'
+require './classroom'
+require './book_storage'
+require './person_storage'
+
+class App
+  def initialize
+    @book_storage = BookStorage.new
+    @rentals = []
+    @person_storage = PersonStorage.new
+  end
+
+  def handle_action(option)
+    case option
+    when '1'
+      @book_storage.list_books
+    when '2'
+      @person_storage.list_people
+    when '3'
+      @person_storage.create_person
+    when '4'
+      @book_storage.create_book
+    when '5'
+      create_rental
+    when '6'
+      list_rentals
+    else
+      puts 'That is not a valid option'
+    end
+  end
+
+  private
+
+  def create_rental
+    puts 'Select a book from the following list by number'
+    @book_storage.each_with_index { |book, i| puts "#{i}) #{book}" }
+
+    book_i = gets.chomp.to_i
+
+    puts
+    puts 'Select a person from the following list by number (not ID)'
+    @person_storage.each_with_index { |person, i| puts "#{i}) #{person}" }
+
+    person_i = gets.chomp.to_i
+
+    puts
+    print 'Date: '
+    date = gets.chomp
+
+    rental = Rental.new(date, @person_storage.get_person_at_index(person_i), @book_storage.get_book_at_index(book_i))
+    @rentals.push(rental)
+
+    puts 'Rental created successfullyy'
+  end
```

Comment:
- [ ] good job cleaning up the main.rb file 👍 in this method text is printed to the console information is gathered and the gathered information is processed, this is not according to the single responsibility rule, try to split this in smaller easier to manage methods. for reference have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid).

Reply:
Thank, you I've addressed this